### PR TITLE
Upgrade openshift/api to release-4.15 branch

### PR DESCRIPTION
**Pull request by julio-lopez**
https://github.com/openshift/api/tree/release-4.15
https://github.com/openshift/client-go/tree/release-4.15

## Pull request type

- [x] Dependency upgrade

## Test Plan

- [x] :zap: CI

### CODE REVIEW

**Comment by julio-lopez** [comment]
#1987

**Comment by julio-lopez** [comment]
@viveksinghggits @pavannd1 Please DO NOT merge yet, it is not complete. I'd like to include an upgrade for Openshift's client-go module as well in the same PR.
Nevertheless, PTAL and LMK if there are any concerns.

Also, what is the plan for upgrading Openshift going forward if automatic upgrades don't work reliably?

**Comment by julio-lopez** [comment]
Also, this includes a couple of minor upgrades to k8s dependencies. Those can be pushed separately in case we need to revert this change.

File: go.mod [hunk 1]
```diff
@@ -40,8 +40,8 @@ require (
 	github.com/onsi/gomega v1.24.1
 
 	//pinned openshift to release-4.5 branch
```
Comment (viveksinghggits) [inline go.mod:42]:
```suggestion
	//pinned openshift to release-4.15 branch
```

File: go.mod [hunk 2]
```diff
@@ -54,13 +54,13 @@ require (
 	gopkg.in/tomb.v2 v2.0.0-20161208151619-d5d1b5820637
 
 	//pinned k8s.io to v0.26.1 tag
-	k8s.io/api v0.26.1
+	k8s.io/api v0.26.3
 	k8s.io/apiextensions-apiserver v0.26.1
-	k8s.io/apimachinery v0.26.1
+	k8s.io/apimachinery v0.26.3
```
Comment (viveksinghggits) [inline go.mod:59]: If we are changing one k8s deps I think we should change others as well. For example `apiextensions-apiserver`, `client-go`, cli-runtime` etc. 
What do you think?

Comment (julio-lopez) [reply go.mod:59]: These were automatically pulled when the openshift/client-go package was updated.
It actually makes sense to put all of the k8s deps in a separate PR altogether. Let me create a separate PR for the k8s deps.

File: go.mod [hunk 3]
```diff
@@ -39,9 +39,9 @@ require (
 	github.com/onsi/ginkgo/v2 v2.6.0
 	github.com/onsi/gomega v1.24.1
 
-	//pinned openshift to release-4.5 branch
-	github.com/openshift/api v0.0.0-20200526144822-34f54f12813a
-	github.com/openshift/client-go v0.0.0-20200521150516-05eb9880269c
+	//pinned openshift to release-4.15 branch
```
Comment (julio-lopez) [inline go.mod:42]: This is technically not 'pinned'
```suggestion
	// use openshift from 'release-4.15' branch
```

**Review by pavannd1 — APPROVED** [review]

**Pull request merged**